nixos/lib/testing/macos-host.nix

```diff
@@ -0,0 +1,18 @@
+{ config, lib, hostPkgs, ... }:
+
+{
+  config = lib.mkIf hostPkgs.stdenv.hostPlatform.isDarwin {
+    defaults = {
+      # On non-darwin, mounting the whole nix store from the host is the default because
+      # this generally makes rebuilds of the VMs faster and the VMs smaller, while reducing
+      # disk I/O.
+      # However, on macOS it currently leads to many crashing guest processes.
+      virtualisation.useNixStoreImage = true;
```

Comment:
I actually cannot reproduce this problem on a small test VM, so I'm not sure if I should keep this option (which I cargo-culted from https://github.com/NixOS/nixpkgs/pull/261694)

Reply:
Had this on an intel mac. I can try it again on my new apple silicon.

In the mean time, please try again with a big VM. I observed so many random crashes.

Reply:
So I tested this using a variation of the example from [How to use NixOS for lightweight integration testing](https://www.haskellforall.com/2020/11/how-to-use-nixos-for-lightweight.html):

<details>
<summary>Test code</summary>

```nix
{ inputs.nixpkgs.url = "/Users/gabriella/proj/nixpkgs";

  outputs = { nixpkgs, ... }: {
    checks.aarch64-darwin.default =
      let
        database      = "postgres";
        schema        = "api";
        table         = "todos";
        username      = "authenticator";
        password      = "mysecretpassword";
        webRole       = "web_anon";
        postgrestPort = 3000;

      in
        nixpkgs.legacyPackages.aarch64-darwin.nixosTest {
          name = "test";

          nodes = {
            server = { config, pkgs, ... }: {
              environment.defaultPackages = [ pkgs.curl ];

              # Open the default port for `postgrest` in the firewall
              networking.firewall.allowedTCPPorts = [ postgrestPort ];

              services.postgresql = {
                enable = true;

                initialScript = pkgs.writeText "initialScript.sql" ''
                  create schema ${schema};

                  create table ${schema}.${table} (
                    id serial primary key,
                    done boolean not null default false,
                    task text not null,
                    due timestamptz
                  );

                  insert into ${schema}.${table} (task) values
                    ('finish tutorial 0'), ('pat self on back');

                  create role ${webRole} nologin;

                  grant usage on schema ${schema} to ${webRole};
                  grant select on ${schema}.${table} to ${webRole};

                  create role ${username} inherit login password '${password}';
                  grant ${webRole} to ${username};
                '';
              };

              users = {
                mutableUsers = false;

                groups."${username}" = { };

                users = {
                  root.password = "";

                  "${username}" = {
                    isSystemUser = true;
                    group = "${username}";
                  };
                };
              };

              systemd.services.postgrest = {
                wantedBy = [ "multi-user.target" ];

                after = [ "postgresql.service" ];

                script =
                  let
                    configuration = pkgs.writeText "tutorial.conf" ''
                      db-uri = "postgres://${username}:${password}@localhost:${toString config.services.postgresql.port}/${database}"
                      db-schema = "${schema}"
                      db-anon-role = "${username}"
                    '';

                  in
                    ''
                      ${pkgs.postgrest}/bin/postgrest ${configuration}
                    '';

                serviceConfig.User = username;
              };
            };

            client = { };
          };


          testScript =
            ''
            import json
            import sys

            start_all()

            server.wait_for_open_port(${toString postgrestPort})

            expected = [
                {"id": 1, "done": False, "task": "finish tutorial 0", "due": None},
                {"id": 2, "done": False, "task": "pat self on back", "due": None},
            ]

            actual = json.loads(
                client.wait_until_succeeds(
                    "curl --fail http://server:${toString postgrestPort}/${table}"
                )
            )

            if expected != actual:
                sys.exit(1)
            '';
      };
  };
}
```
</details>

… and that test succeeds without issues.  That's running `postgres` and `Postgres`.  If you have a test of your own that was failing before without the workaround you can try running that using this branch and let me know if you're still able to reproduce the problem.  However, currently I'm not able to reproduce the problem.